### ISSUE TITLE
Django : attempt to write a readonly database

### ISSUE THREAD

**Issue by mfoacs** [nginx/unit#163]
Here is my setup:
- unit version: 1.4, configured as ./configure --prefix=/usr/local --user=www --group=www
- nginx version: nginx/1.14.0
- Django 2.1.1 but also Django 1.8
- sqlite3 (development server)
- python 3.6.6
- Freebsd 11.2-CURRENT-p2 amd64

The process is launched and running:
`sudo /usr/local/sbin/unitd --user www --group www --log /usr/local/unit.log` OR
`sudo /usr/local/sbin/unitd --user www --group www --control http://127.0.0.1:8443/ --log /usr/local/unit.log`
(Both methods work and produce the same result)

I can browse the site structure without issues but all attempts to change the database fail:

```
Request Method: POST
OperationalError attempt to write a readonly database.
```

However,
`waitress-serve --listen=0.0.0.0:8400 myapp.wsgi:application` works as expected.

Any ideas?

**Comment by VBart** [comment]
Thanks for the report. Could you provide an example app? We should try to reproduce.

**Comment by mfoacs** [comment]
Yes, here is a very basic setup:

1. `virtualenv venv & source venv/bin/activate`
2. `pip install Mezzanine`
3. `mezzanine-project testunit & cd testunit `
4. `python manage.py migrate`
5. `python manage.py createsuperuser`

The json file:
```
{
    "listeners": {
        "*:8500": {
            "application": "testunit"
        },
    },

    "applications": {
        "testunit": {
            "type": "python 3.6",
            "processes": 2,
            "working_directory": "/home/mda/testunit",
            "home": "/home/mda/venv",
            "user": "www",
            "group": "www",
            "path": "/home/mda/testunit",
            "module": "testunit.wsgi"
        },

    },
    "access_log": "/var/log/access.log"
}
```

Start unit and configure the app:

`sudo /usr/local/sbin/unitd --user www --group www --control 127.0.0.1:8443`
```
sudo curl -X PUT -d@/home/mda/json/sites.json http://127.0.0.1:8443/
{
	"success": "Reconfiguration done."
}
```
Connect with nginx
```
upstream unit_backend {
        server 127.0.0.1:8500;
}

server {
    # the port your site will be served on
    listen 80 default_server;
    # the domain name it will serve for
    server_name testunit ;
    charset     utf-8;

    # max upload size
    client_max_body_size 75M;   # adjust to taste

    location ~ \.php$ {
            deny all;
    }

    # Django media
    location /templates  {
        alias /home/mda/testunit/templates;
    }

    location /static {
        alias /home/mda/testunit/static;
    }

    location / {

                proxy_pass         http://unit_backend;
                proxy_set_header   Host $host;

        }

         
}
```
Restart nginx and go to your site's admin. Try to login and you should have the error:
```

Request Method: | POST
-- | --
http://testunit/admin/login/?next=/admin/
1.11.15
OperationalError
attempt to write a readonly database
/home/mda/venv/lib/python3.6/site-packages/django/db/backends/sqlite3/base.py in execute, line 328
/usr/home/mda/venv/bin/python3
3.6.5
['/home/mda/testunit',  '/home/mda/venv/lib/python36.zip',  '/home/mda/venv/lib/python3.6',  '/home/mda/venv/lib/python3.6/lib-dynload',  '/usr/local/lib/python3.6',  '/home/mda/venv/lib/python3.6/site-packages']
Tue, 11 Sep 2018 06:38:58 +0000
```

**Comment by iurisilvio** [comment]
User `www` has write permission to this sqlite database?

When you run it with waitress, I think it is using another user.

**Comment by ghostinushanka** [comment]
Apologies for apparent necroposting, I have seen today what appears to be same behavior and this issue is not closed. A question for @mfoacs 

> I have also tried running unit as the same user as the database owner. Or changing the owner of the directory tree to the same user running the unitd daemon.

Based on the configuration information provided, are you positive that the `www` user/group does not only have the write permissions on the sqlite database file, but also to the directory that encompasses it? Your working directory is `/home/mda/testunit`, I would therefore assume, that user `mda` exists on the system. For correct operation, either user `www` or group `www` should also have write permissions on the `/home/mda` directory (alternatively everybody)

**Comment by mfoacs** [comment]
Yes, that's correct.
Running unitd with my local (and existing) user solves the issue. But that
will obviously become an issue for any other users.
Giving write permissions to everybody it's not an option.

**Comment by mar0x** [comment]
In your original post you run `waitress-serve` from current user (I assume it is `mda`). So you need to change your Unit configuration to run `testunit` application as `mda` user and corresponding group (check the primary group of `mda` user).

**Comment by mfoacs** [comment]
Indeed you're right.
However, that was the reason why I made the post in the first place: I ran
into trouble when I configured the testunit app to run as *mda:mda*, but
the unit deamon was running as *www:www*, and although the user mda was in
the group www, things didn't seem to click.
I have started from scratch and instead of compiling unit from sources I've
installed the binary package and it works as expected.

**Comment by mar0x** [comment]
Of course,  `unitd` need to be started as `root` to be able to run application as other arbitrary user.
